docker/centos7.3/Dockerfile

```diff
@@ -0,0 +1,148 @@
+FROM docker.io/centos:7.3.1611
+MAINTAINER Hortonworks
+
+ENV container docker
+ENV TERM linux
+ENV PS1 "[\u@cloudbreak \W]# "
+
+ENV SALT_REPO_URL=https://repo.saltstack.com/yum/redhat/salt-repo-2017.7-1.el7.noarch.rpm
+ENV SYSTEMCTL_REPLACEMENT=https://raw.githubusercontent.com/tibkiss/docker-systemctl-replacement/06367e13d96fed16447c2ed6dd1979d559e7f9a5/files/docker/systemctl.py
+
+# Fix the default shell from dash to bash
+# Do it one line. Otherwise, after doing rm, docker build cannot find a shell to
+# run more RUN commands :)
+RUN bash -c 'rm /bin/sh; ln -s /bin/bash /bin/sh'
+RUN chsh -s /bin/bash
+
+#################################################################################
+# Systemd stuff
+#################################################################################
+RUN (cd /lib/systemd/system/sysinit.target.wants/; for i in *; do [ $i == systemd-tmpfiles-setup.service ] || rm -f $i; done); \
+    rm -f /lib/systemd/system/multi-user.target.wants/*;\
+    rm -f /etc/systemd/system/*.wants/*;\
+    rm -f /lib/systemd/system/local-fs.target.wants/*; \
+    rm -f /lib/systemd/system/sockets.target.wants/*udev*; \
+    rm -f /lib/systemd/system/sockets.target.wants/*initctl*; \
+    rm -f /lib/systemd/system/basic.target.wants/*;\
+    rm -f /lib/systemd/system/anaconda.target.wants/*;
+#################################################################################
+
+#################################################################################
+# Setup yum repos
+#################################################################################
+
+# XXX: Disabled for public build
+# Disable fast-mirrors plugin and force everything through our internal repo
+#RUN sed -i -e 's/^enabled=1/enabled=0/g' /etc/yum/pluginconf.d/fastestmirror.conf
+
+# Install epel repo files. epel-release has to go first before some of the other
+# packages can be installed
+RUN yum -y install epel-release
+
+# XXX: Disabled for public build
+# BUG-68936 Fixing the image to use the HWX internal repos to avoid massive
+# downloads due to cache-misses on the internal caching-proxy.
+#RUN rm -rf /etc/yum.repos.d/*
+#COPY files/hwx-internal.repo /etc/yum.repos.d/
+
+# Give preference to our repos
+RUN yum -y install yum-priorities
+#################################################################################
+
+#################################################################################
+# Install basic OS packages + GCC (to build bootstrap)
+#################################################################################
+RUN yum install -y yum-utils yum-plugin-ovl tar git curl bind-utils \
+                   unzip wget gcc net-tools sudo zip lsof strace redhat-lsb
```

Comment:
I would not install gcc and all the staff required to compile suid binary.

Multi stage build:
https://docs.docker.com/engine/userguide/eng-image/multistage-build/

Reply:
Let's consider this at a later version